conans/client/graph/graph_builder.py

```diff
@@ -148,6 +148,7 @@ def _handle_require(self, name, node, require, dep_graph, check_updates, update,
             new_node.inverse_closure.add(node)
             node.public_deps[new_node.name] = new_node
 
+            new_node.build_require = node.build_require or require.build_require
```

Comment:
I had to postpone https://github.com/conan-io/conan/issues/4738 for 1.16, but please, at least put a comment if the code is new. That is a new corner case that it is impossible for anyone except you to know why. What does it mean that `node.build_require` is None and why it should take it from `require`? What is `require`?

Reply:
Done